File: backends/aspnet-core/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace LogisticsAspNetCore.Models;

public class Vehicle
{
    public long Id { get; set; }

    [Required]
    public string RegistrationNumber { get; set; } = string.Empty;

    [Required]
    public string Type { get; set; } = string.Empty;

    [Range(0.01, double.MaxValue)]
    public double CapacityKg { get; set; }

    public bool Available { get; set; }
}

File: backends/laravel/app/Models/Vehicle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Vehicle extends Model
{
    protected $fillable = [
        'registration_number',
        'type',
        'capacity_kg',
        'available',
    ];

    protected $casts = [
        'capacity_kg' => 'float',
        'available' => 'boolean',
    ];
}
